Makefile

```diff
@@ -222,14 +238,17 @@ CONFIG_FILE_PREFIX=.Makefile.config$(BUILD_SUFFIX).
 CONFIG_FILE=$(CONFIG_FILE_PREFIX)$(CONFIG_HASH)
 
 .PHONY: mismatched_config_file_check
-mismatched_config_file_check:
+mismatched_config_file_check: | $(BUILD_DIR)/.
+	@echo '  '
 	$(BRIEF_PARTIAL) Checking for mismatched config...
 	$(COMMAND) ls $(CONFIG_FILE_PREFIX)* 2>/dev/null | grep -v $(CONFIG_FILE) | xargs -ri sh -c 'echo "Mismatched config file \"{}\" detected.  You need to \"make clean\"."; false'
-	$(BRIEF) No mismatched config found
-
+	$(BRIEF) No mismatched config found in $(CONFIG_FILE)
 
-$(CONFIG_FILE): | mismatched_config_file_check
+# Search for CC also in 'env' command below, while storing config to a file.
+# Otherwise, 'make' may fail if user's environment has no BUILD_ env-vars set.
+$(CONFIG_FILE): | $(BUILD_DIR)/. mismatched_config_file_check
 	$(BRIEF) Saving config to $@
+	$(COMMAND) env | grep -E "BUILD_|CC"         >> $@
```

Comment:
Am dumping any `BUILD_` env-vars to the config file. Need to pull-in `CC` with this `grep`, otherwise if no `BUILD_` env-vars are send, this `grep` will fail. That causes whole `make` to fail.

Only question is: Do we still need L 252 below? I think it's ok to leave both ... doesn't hurt anything.